docs/xks/operator-guide/architecture-and-design.md

```diff
@@ -0,0 +1,117 @@
+---
+id: architecture-and-design
+title: Architecture and design
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+## Overview
+
+[Xenit Kubernetes Framework](https://github.com/search?q=topic%3Axenit-kubernetes-framework+org%3AXenitAB+fork%3Atrue) (XKF) is the open source building blocks for a service Xenit AB provides customers: [Xenit Kubernetes Service](https://xenit.se/it-tjanster/kubernetes-eng/) (XKS)
+
+In the perspective of [Microsoft Cloud Adoption Framework](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/enterprise-scale/architecture) (CAF), Xenit Kubernetes Service is an enterprise-scale landing zone.
+
+The primary goal of the framework is to provide a standardized service to customers where the creators of the framework (employees at Xenit) can manage multiple environments without having to be the one who set it up - because it looks just about the same in all places. This also provides the ability to fix issues, change standards in one place and not having to repeat the same thing over and over at all customer (or two engineers solving the same problem at two different customers).
+
+<img alt="XKS Overview" src={useBaseUrl("img/assets/xks/operator-guide/aks-overview.jpg")} />
+
+## Role-based access management
+
+All role-base access control (RBAC) and identity & access management (IAM) is handled with Azure AD. Azure AD groups are created and nested using the framework and framework admins as well as customer end users are granted access through these different groups.
+
+Everything, where possible, exposes two different permissions: Reader and Contributor
+
+These permissions are scoped in many different ways and start at the management group level, subscription level, resource group level and at last namespaces in Kubernetes. These are also split over the different environments (development, quality assurance and production) meaning you can have read/write in one environment but only read in the others.
+
+An owner role and group is also created for most resources, but the recommendation is not to use it as owners will be able to actually change the IAM wich in most cases is undesirable.
+
+The normal customer end user (often an engineer / developer, referred to as a tenant) is granted read/write to their resource groups and namespaces, meaning they will be able to add/remove whatever they want in their limited scope. This usually means creating deployments in Kubernetes as well as databases and other stateful resources in their Azure Resource Groups. When using AWS Elastic Kubernetes Service (EKS) the delegation insn't as rigorous as in Azure and the default setup creates three accounts where all the customer tenants share resources.
```

Comment:
The tenant concept should probably be introduced in the overview?
Perhaps "Azure resource groups and Kubernetes namespaces" for clarity? (If so, the following sentence can be symmetrically simplified.)

Reply:
done!